magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,417 @@
+        wl = dl1_params["width"][index] / dl1_params["length"][index]
+        log_intensity = np.log10(dl1_params["intensity"])[index]
+        timestamp = dl1_params["timestamp"][index]
+        time_diff = dl1_params["time_diff"][index]
+        n_islands = dl1_params["n_islands"][index]
+        n_pixels = dl1_params["n_pixels"][index]
+        signal_pixels = dl1_params["n_pixels"][index]
+
+        trans = 0.58  # transmission of the cloud
+        trans = trans ** (1 / np.cos(zenith))  # .to_value("")
+        nlayers = 10  # to simplify calculations with the model cloud is considered to be composed of multiple layers
+        Hcl = np.linspace(Hc, Hc + dHc, nlayers)  # position of each layer
+        transl = trans_height(Hcl, Hc, dHc, trans)  # transmissions of each layer
+        transl = np.append(
+            transl, transl[-1]
+        )  # add one more element for heights above the top of the cloud
+
+        alt_rad = np.deg2rad(dl1_params["alt"][index])
+        az_rad = np.deg2rad(dl1_params["az"][index])
+
+        eff_focal = focal_eq[telid]  # *u.m # m
+        m2deg = np.rad2deg(1) / eff_focal * u.degree
+
+        impact = dl1_params["impact"][index] * u.m
+        psi = dl1_params["psi"][index] * u.deg
+        cog_x = (dl1_params["x"][index] * m2deg).value * u.deg
+        cog_y = (dl1_params["y"][index] * m2deg).value * u.deg
+
+        # Source position
+        pointing_altaz = SkyCoord(alt=alt_rad * u.rad, az=az_rad * u.rad, frame=AltAz())
+        telescope_pointing = SkyCoord(
+            alt=dl1_params["pointing_alt"][index] * u.rad,
+            az=dl1_params["pointing_az"][index] * u.rad,
+            frame=AltAz(),
+        )
+
+        tel_frame = TelescopeFrame(telescope_pointing=telescope_pointing)
+        tel = pointing_altaz.transform_to(tel_frame)
+
+        src_x = tel.fov_lat.to(u.deg).value
+        src_y = tel.fov_lon.to(u.deg).value
+
+        # Transform to Engineering camera
+        src_x, src_y = -src_y * u.deg, -src_x * u.deg
+        cog_x, cog_y = -cog_y, -cog_x
+
+        pix_x_tel = (camgeom[telid].pix_x * m2deg).to(u.deg)
+        pix_y_tel = (camgeom[telid].pix_y * m2deg).to(u.deg)
+
+        distance = np.abs(
+            (pix_y_tel - src_y) * np.cos(psi) + (pix_x_tel - src_x) * np.sin(psi)
+        )
+
+        d2_cog_src = (cog_x - src_x) ** 2 + (cog_y - src_y) ** 2
+        d2_cog_pix = (cog_x - pix_x_tel) ** 2 + (cog_y - pix_y_tel) ** 2
+        d2_src_pix = (src_x - pix_x_tel) ** 2 + (src_y - pix_y_tel) ** 2
+
+        distance[d2_cog_pix > d2_cog_src + d2_src_pix] = 0
+        dist_corr_layer = model2(impact, Hcl, zenith) * u.deg
+
+        ilayer = np.digitize(distance, dist_corr_layer)
+        trans_pixels = transl[ilayer]
+
+        inds_img = np.where(
+            (dl1_images["event_id"] == event_id)
+            & (dl1_images["tel_id"] == telid)
+            & (dl1_images["obs_id"] == obs_id)
+        )[0]
+
+        if len(inds_img) > 0:
```

Comment:
this condition should always be true if we have images in the files. 
you should add at least some error message if you find an event without image rather then skipping it silently

Reply:
Sure, the following line is added:

else:
    raise ValueError("Error: 'inds_img' list is empty!")